### DISCUSSION TITLE
PlayWright wait times & Input field times

### DISCUSSION THREAD

**Discussion by pditmar** [artilleryio/artillery#3323] [Artillery]
We have a lot of Playwright functional Page Object Model test scripts which work great in Artillery.

Some of our methods contain wait times and also typing time. Is there any way to automatically remove those from the reports/metric transaction reponse times that I create steps for.

The following is an example, multiple actions are taken before the submit :

```
  const enterBasicData = async (dateOfBirth: string, durationTerm: string) => {
    await page.getByPlaceholder('DD-MM-JJJJ').fill(dateOfBirth)
    await page.getByText(durationTerm).first().click()
    await page.locator('#btnSelect-Basis').click()
    await page.getByRole('button', { name: 'Submit' }).click()
  }
```

Called like this : 

`await onlinePage(page).enterBasicData('01-01-1981', '65')`

Another thing they do is create a delay function with a timeout (JS) or use the page.waitForTimeout

Do I have to instruct the teams to re-write their scripts and split everything up? Or are there other possibilities?

**Comment by hassy** [comment]
Hi @pditmar, no not at the moment. The duration of the step takes everything into account.

Are you looking to make metrics generated by different runs more consistent / comparable?

**Reply by pditmar** [reply]
For browser driven performance tests we are usually only interested in the response times of the screen changes or for example until a page can be interacted with. As you can see in the example the teams within my company are adding multiple actions that I don't want to measure within enterBasicData. 

Our SLA could be for example 95 percentile within 1 second from submitting the data until the next page is loaded. If the filling of fields is included in the "await step" response time we **might** not reach the SLA due to the time wasted typing text and selecting object before pressing submit.

I simply want to measure from the Submit button click until the next page is loaded. So either I need to instruct them to split up the button press from the rest of the actions. Or I need to find a way to extract the typing and object selection times from the response time.
